pkg/kvstore/etcd_lease.go

```diff
@@ -94,7 +102,7 @@ func (elm *etcdLeaseManager) GetLeaseID(ctx context.Context, key string) (client
 			elm.keys[key] = elm.current
 			elm.mu.Unlock()
 
-			return elm.current, nil
+			return info.session, nil
```

Comment:
We should check here if sessions in not Orphaned I think.
in https://github.com/cilium/cilium/pull/28834/files#diff-85b5ad3b5a6b7004aafad37cdbb264e395fd491231f4c8cae8ce88554e7d1145R193
we can Orphan session, but it potentially can still be in elm.leases as we release lock after that and remove it later on in `waitForExpiration`

Reply:
I was thinking about something like this, to make it less likely to happen:
```
select {
case <-info.session.Done():
    continue
default:
}
```

Reply:
The main issue is that we cannot completely get rid of the uncertainty period, due to the delay between server-side expiration and client-side detection (with consequent removal of the session from the list of known ones). While additional checks could help slightly reducing that window, they also make the logic more complex (as they would break the assumption that sessions get GCed only by the `waitForExpiration` function). For this reason, the idea here is to essentially rely on client-side retries in the unlikely case that the session is expired. 

As discussed offline, I've extended the comments of the GetLease/GetSession functions for additional clarity about that assumption: https://github.com/cilium/cilium/pull/28834/commits/48518d6781c67198f62fbd0db094c820309bf9cf